vital/types/homography.h

```diff
@@ -162,21 +156,22 @@ class VITAL_EXPORT homography_ :
    *
    * \return New homography transformation instance.
    */
-  virtual homography_sptr normalize() const;
+  homography_sptr normalize() const override;
 
   /// Get a new \p homography that has been inverted.
   /**
-   * \throws non_invertible_matrix When the homography matrix is non-invertible.
+   * \throws non_invertible_transform
+   *   When the homography matrix is non-invertible.
    * \return New homography transformation instance.
    */
-  virtual homography_sptr inverse() const;
+  transform_2d_sptr inverse() const override;
```

Comment:
the need to static cast back to a homography is really annoying because inverse is a common operation on homographies and we almost always want the result to be another homography.  Maybe we should add a `inverse_homography()` function that returns a homography and have the virtual `inverse()` function call that.

Reply:
Ah, the woes of polymorphism and the lack of covariant returns...